matrix/main.py

```diff
@@ -132,6 +132,13 @@ def setup(matrix, args=None):
     parser.add_argument("-k", "--keep-models", action="store_true",
                         default=False, help="Keep the created per-test models "
                                             "for further inspection")
+    parser.add_argument("-r", "--deployer", default="python-libjuju",
+                        help="Method to use to deploy charms. Defaults to "
+                        "'python-libjuju'. Other options include 'congure-up'."
+                        " Note that if you choose conjure-up, --path "
+                        "should point to a conjure-up spell "
+                        "(with metadata.yaml) "
+                        "instead of to a vanilla juju bundle.")
```

Comment:
A spell is a directory with a metadata.yaml, and either a 'bundle-location' key in that file, *or* a separate bundle.yaml file next to it. The only required keys in metadata.yaml are 'friendly-name' and 'version'. I think just checking for a metadata.yaml with a 'friendly-name' key would do for matrix's purposes, but maybe I'm missing something?

Also - line 137 has a typo - "congure-up".